### ISSUE TITLE
[Bug]: Executing a command with '--grep @tags' overrides the 'workers: 1,' from playwright.config.js

### ISSUE THREAD

**Issue by THSkillwork** [microsoft/playwright#32449]
### Version

1.46.1

### Steps to reproduce

1. Tag multiple tests with any tag - e.g. @smokeCheck
2. Set the number of workers in your playwright.config.js to 1 - e.g.

 import { defineConfig } from '@playwright/test';

export default defineConfig({
  workers: 1,
  testMatch: 'test.spec.js',
});
3. Set the fullyParallel configuration to 'false' in the playwright.config.js
/* Run tests in files in parallel */
  fullyParallel: false,

4. Execute a command including only the tagged tests - npx playwright test --grep @smokeCheck --project chromium

### Expected behavior

I expect the tests to always use 1 worker when it is specifically configured to do so in the playwright.config.js

### Actual behavior

Executing a test command with @tags, overrides the  playwright.config.js and executes as many tests in parallel as possible

### Environment

```shell
System:
  OS: Windows 11 
  CPU: 13th Gen Intel(R) Core(TM) i5-1335U   1.30 GHz
Binaries:
  node - v22.1.0
  npm - 10.8.2
npmPackages:
      "@playwright/test": "^1.46.1"
```

**Comment by THSkillwork** [comment]
I just set the workers directly in the execution command like:
npx playwright test --grep @smokeCheck --project chrome --workers 1
But it is still annoying we have to do that explicitly

**Comment by pavelfeldman** [comment]
Works for me, I think your issue is that you are running it from a different location and config is not picked up. Please file a new issue with a full repo as required by the BUG template so that we could reproduce it locally.
